packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -220,13 +220,13 @@ const generateConfig = (
 
     //This could handled by another workflow instead of the application workflow
     closing: {
-      entry: [actions.assignRequestId],
+      entry: [actions.assignRequestId, actions.sendCloseChannelResponse],
```

Comment:
This might be a bad idea because (I think) the second action depends on first completing?

Reply:
I think it works but that doesn't make it a good idea. (I think the `interpreter` currently exported by xstate bumps `assign` actions to first in the queue.)

`assignRequestId` should probably happen as an `exit` action in the transition _into_ `closing`.

That's more clear, because the event is right there, meaning it's obvious that the event has a `requestId` property.

And, it's more robust, because we know that the request ID is assigned to the context before the `sendCloseChannelResponse`

## Alternative

The `sendCloseChannelResponse` is _important_, so maybe it should be a _service_, not an _action_. So, maybe when you enter the `closing` state, the first thing you do is
`{ invoke: { src: 'sendCloseChannelResponse', onDone: 'doTheNextThing' } }`

Reply:
Thanks for this @andrewgordstewart . Pairing with @tomclose, we refactored so that these actions happen either side of signing the `isFinal` state. We also allowed for the request to get an error response if you made the API call when it was not your turn.